shipctl/x86_64/WindowsServer_2016/utility.ps1

```diff
@@ -1130,7 +1193,75 @@ function _send_airbrake_notification() {
 
   Write-Output "Requesting Airbrake project: $r_project_id"
 
-  _send_web_notification -payload "$env:opt_payload" -auth "$r_authorization" -endpoint "$r_endpoint" -security_protocol "tls12"
+  _send_web_notification -payload "$env:opt_payload" -auth $r_authorization -endpoint "$r_endpoint" -security_protocol "tls12"
+}
+
+function _send_newrelic_notification() {
+  $appId = ""
+  $r_endpoint = ""
+  $default_post_deployment_payload = '{"$opt_type":{"revision":"$opt_revision","description":"$opt_description","user":"$opt_username","changelog":"$opt_changelog"}}'
+  $default_get_appid_payload = @{ "filter[name]" = "$env:opt_appName" }
+  $default_get_payload = ""
+  $default_post_payload = ""
+  $authorization = $(get_integration_resource_field "$env:opt_resource" token)
+
+  if ($authorization) {
+    $r_authorization = @{ "X-Api-Key" = "$authorization" }
+  } else {
+    $r_authorization = @{}
+  }
+
+  if (!$env:opt_username) {
+    $env:opt_username = "Shippable"
+  }
+
+  $r_url=$(get_integration_resource_field "$env:opt_resource" url)
+  if (!$r_url) {
+    throw "Error: no url found in resource $env:opt_resource"
+  }
+
+  if (!$env:opt_appId -and !$env:opt_appName) {
+    throw "Error: --appId or --appName should be present in shipctl notify"
+  }
+
+  $appId=$env:opt_appId
+  if (!$appId) {
+    $r_endpoint="$r_url/applications.json"
+    $default_get_payload=$default_get_appid_payload
+    $applications=$(_get_web_request -payload $default_get_payload -auth $r_authorization -endpoint "$r_endpoint") | ConvertFrom-Json
+    $appId=$applications.applications[0].id
+  }
+
+  if (!$appId) {
+    throw "Error: Unable to find an application on NewRelic"
+  }
+  $r_endpoint="$r_url/applications/$appId/deployments.json"
+  $default_post_payload="$default_post_deployment_payload"
+
+  if ($env:opt_payload) {
+    if (!(Test-Path $env:opt_payload)) {
+      throw "Error: file not found at path: $env:opt_payload"
+    }
+    try {
+      $result = Get-Content $env:opt_payload -Raw | ConvertFrom-Json -ErrorAction Stop
+    } catch {
+      throw "Error: payload is not valid JSON"
+    }
+  } else {
+    if (!$env:opt_type) {
+      throw "Error: --type is missing in shipctl notify"
+    }
+    if (!$env:opt_revision) {
+      throw "Error: --revision is missing in shipctl notify"
+    }
+    Out-File -FilePath "$env:TEMP/payload.json" -InputObject $default_post_payload
+    $env:opt_payload = "$env:TEMP/payload.json"
+    shipctl replace $env:opt_payload
+  }
+
+   Write-Output "Recording deployments on NewRelic for appID: $appId"
+
+  _send_web_notification -payload "$env:opt_payload" -auth $r_authorization -endpoint "$r_endpoint"
```

Comment:
We're doing `-security_protocol "tls12"` for AirBrake. Is that not required here?

Reply:
Nope, new relic seems to work without `-security_protocol "tls12"`